### DISCUSSION TITLE
OV2640 camera module on ESP32-WROVER board with tasmota

### DISCUSSION THREAD

**Discussion by whabby** [arendst/Tasmota#21053] [Support request]
Hello everyone

For my project I would like to control four relays with Tasmota. Further on I need a SR04 distance sensor and for visual control I would like to set up a camera. I have planned the following hardware:

- ESP32-WROVER board (https://de.aliexpress.com/item/1005005466779730.html)
- SR04 distance sensor (https://de.aliexpress.com/item/1005006320348756.html)
- OV2640 camera module 2MP (https://de.aliexpress.com/item/33046344720.html)
I would like to use this camera module as it has much better optics than the original camera.

I have flashed my customised version of Tasmota 13.03 on the ESP and everything seems to work perfectly, but only with the original camera connected to the camera port. However, my OV2640 camera module 2MP has differently labelled connections. 
  SCL (System Clock ??) SDA (System Data ??) and DCLK (Data Clock ??) are labels that I cannot find in the Tasmota camera configuration.

As far as I know, the OV2640 should be compatible with Tasmota. I really hope that someone can help me with my connection problems.

Many thanks in advance!
Greetings Whabby

**Comment by Jason2866** [comment]
See here as reference https://github.com/espressif/esp32-camera/blob/bae46be5eb126c46c692a29a953991117f68336b/examples/camera_example/main/take_picture.c#L100-L116

You have to use a ESP32 module with PSRAM. Do not use a ESP32 Cam board if you want to use the extra CAM module.
